jetty-ee9/jetty-ee9-nested/src/main/java/org/eclipse/jetty/ee9/nested/ServletCoreResponse.java

```diff
@@ -137,12 +137,15 @@ public void write(boolean last, ByteBuffer byteBuffer, Callback callback)
             last = false;
         try
         {
-            if (!_wrapped && !_baseResponse.isWritingOrStreaming())
+            if (last && !_wrapped && !_baseResponse.isWritingOrStreaming())
             {
+                // We can bypass the HttpOutput stream, but we need to update its bytes written
+                _baseResponse.getHttpOutput().setBytesWritten(byteBuffer.remaining());
```

Comment:
This seems too early. 
If the write at the line below fails, the bytes would have been updated, but they were not written.

`HttpOutput` was maintaining internally the count of bytes written, and setting it from here would set the non-compressed value, while normally `HttpOutput` would record the compressed ones.

Reply:
That is why we have two counts of bytes written.  The real one is down in HttpChannel and it records bytes going to the wire (after compression etc.).   The count in `HttpOutput` is an API count and is needed to implement the servlet spec behaviour of committing when the number of bytes set in content length have been written.   By its nature, it has to be before the write is attempted, as the write may be different if the count == contentLength.